pygfx/utils/text/_atlas.py

```diff
@@ -0,0 +1,160 @@
+import threading
+
+import numpy as np
+
+from ...resources import Texture
+
+
+# todo: how to invalidate the pipeline of objects that use the atlas when the texture is changed?
```

Comment:
I've been thinking a lot about this, since the lighting system also needs it... I'm imagining some sort of dependency tracking mechanic using weakrefs...

Reply:
Yeah, a more generic approach to solve these use-cases sounds like a good idea.